### ISSUE TITLE
Workspace - User is navigate to thread message when going to #announce room

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51820]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.56-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+ds@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Create a workspace > add 2 members
3. Go to #announce room
4. Send a message > Create a thread message
5. Navigate to Workspace > Click on the three dot menu > Click #announce

## Expected Result:
User is navigated to #announce room main page

## Actual Result:
User is navigated to #announce room thread message page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/9b97bba9-4001-40d6-8c57-803be16c45b2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@garrettmknight  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by garrettmknight** [comment]
@eVoloshchak can you please review this proposal?

**Comment by eVoloshchak** [comment]
We should proceed with @daledah's [proposal](https://github.com/Expensify/App/issues/51820#issuecomment-2451280412), since It makes use of the already-existing `ReportUtils.isThread` method

🎀👀🎀 C+ reviewed!

**Comment by mountiny** [comment]
I agree, I think @Nodebrute already nailed it and it was first, also noted its valid issue for admins room too, can you fix it there too?

**Comment by mountiny** [comment]
@Nodebrute, we are also working on a new process which will require adding unit tests for each bug fix to increase the chances of this issue not getting re-introduced. Can you also add unit tests to cover this case please?

**Comment by mountiny** [comment]
@Nodebrute appreciated

**Comment by mountiny** [comment]
@eVoloshchak can you please complete the checklist here? Thanks!

**Comment by garrettmknight** [comment]
@eVoloshchak bump on the checklist.

@Nodebrute just paid you out.

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/33280
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/33280/files#r1875956571
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not sure if additional discussion is needed, this Is a pretty simple navigation bug
- Determine if we should create a regression test for this bug
Is it easy to test for this bug? Yes
Is the bug related to an important user flow? Yes
Is it an impactful bug? Yes
This isn't a critical bug, but it is pretty impactful and related to widely used functionality, we might add a regression test

**Regression Test Proposal**

1. Create a workspace > add 2 members
2. Go to #announce room
3. Send a message > Reply in thread -> send a message in thread
4. Navigate to Settings/Workspaces > Click on the three-dot menu > Click on `Go to #announce room`
5. Verify user is navigated to the #announce room

**Do we agree 👍 or 👎**

**Comment by garrettmknight** [comment]
Payment Summary:
- Contributor: @Nodebrute $250 paid via Upwork
- Reveiwer: @eVoloshchak $250 via NewDot

**Comment by JmillsExpensify** [comment]
$250 approved for @eVoloshchak
